lib/internal/process/per_thread.js

```diff
@@ -352,17 +352,12 @@ function wrapProcessMethods(binding) {
   }
 
   /**
-   * Loads the `.env` file to process.env.
-   * @param {string | URL | Buffer | undefined} path
+   * Loads the `.env` file onto `process.env`.
+   * @param {string | URL | Buffer | uFndefined} path
    */
   function loadEnvFile(path = undefined) { // Provide optional value so that `loadEnvFile.length` returns 0
-    if (path != null) {
-      getValidatedPath ??= require('internal/fs/utils').getValidatedPath;
-      path = getValidatedPath(path);
-      _loadEnvFile(path);
-    } else {
-      _loadEnvFile();
-    }
+    const loaded = util.loadEnvFile(path);
+    ObjectAssign(process.env, loaded);
```

Comment:
Given how trivial it is to load a .env file onto `process.env` I even wonder if we could deprecate/remove `process.loadEnvFile`.

To me personally it feels much clearer to have this sort of operation more manual and have the user load their .env files with`loadEnvFile` and then use the values to populate `process.env` themselves if and how they want (with any overriding, filtering, etc. logic).